services/navigation/builtin/builtin.go

```diff
@@ -293,15 +294,21 @@ func (svc *builtIn) Reconfigure(ctx context.Context, deps resource.Dependencies,
 		return err
 	}
 
-	// Parse vision services from the configuration
-	var visionServiceNames []resource.Name
+	var obstacleDetectorPairs []motion.ObstacleDetector
 	var visionServices []vision.Service
-	for _, svc := range svcConfig.VisionServices {
-		visionSvc, err := vision.FromDependencies(deps, svc)
+	for _, pbObstacleDetectorPair := range svcConfig.ObstacleDetectors {
+		visionSvc, err := vision.FromDependencies(deps, pbObstacleDetectorPair.VisionService)
 		if err != nil {
 			return err
 		}
-		visionServiceNames = append(visionServiceNames, visionSvc.Name())
+		camera, err := camera.FromDependencies(deps, pbObstacleDetectorPair.Camera)
+		if err != nil {
+			return err
+		}
+		obstacleDetectorPairs = append(obstacleDetectorPairs, motion.ObstacleDetector{
+			VisionService: visionSvc.Name(),
+			Camera:        camera.Name(),
+		})
 		visionServices = append(visionServices, visionSvc)
```

Comment:
[minor] Current paradigm for explore would require using the name of the given visionServices and finding the associated camera via the motion cfg. This will work but it would be preferable to keep the obstacle detector struct in the nav config, replacing visionServices. 

Note: @nfranczak Is there a way we can combine the motionCfg's vision service names and camera names with the full objects stored in the nav config? I'm not sure if there is but wanted to bring up the possiblity.

Reply:
to clarify, do you mean changing the `obstacleDetector` struct so that we also have a `vision.Service` apart from just its name?

I will need to think about this a little more, but a safe solution is not immediately obvious to me.

Curious what others think 👀

Reply:
I'm curious why it wouldn't be save to save the actual vision service and camera (not just their names) in the navigation service. I think this would be fine, since the navigation service depends on those resources, so if those resources reconfigure, the navigation service will also be reconfigured.

Reply:
I agree that we can save the actual vision service and camera (not just their names) in the navigation service. 
To this end we already have `visionServices []vision.Service` within the `builtIn` struct of the navigation service.
This can even be updated to become `[]map[vision.Service][camera.Camera]`.

However, I don't think it would be safe to have the `ObstacleDetector` struct (our go backend representation) become 
```
type ObstacleDetector struct {
	VisionService vision.Service
	Camera           camera.Camera
}
```
because of how we choose to represent an `obstacleDetector` in `motion.proto`.
Specifically, I do not think we can go from a `protobuf resource.Name` to a `vision.Service` or from a `protobuf resource.Name` to a `camera.Camera` when we would be calling `MoveOnGlobe`.

------------------------------------------
What do we think of replacing `visionServices []vision.Service` within the `builtIn` struct of the navigation service with `[]map[vision.Service][camera.Camera]`?

Reply:
I understand the concern about proto compatibility, that's why we have the vision service names in the MotionConfiguration and the full vision service in the nav struct. I would like us to do the same with `ObstacleDetection`. Have an `ObstacleDetectionNames` struct and an `ObstacleDetection` struct. This is similar to your map suggestion, however, I am against using a large object such as a vision service as a key to a map. Id prefer just a `[]ObstacleDetector`

```
type ObstacleDetector struct {
	VisionService vision.Service
	Camera           camera.Camera
}

type ObstacleDetectorName struct {
	VisionServiceName string
	CameraName           string
} // Note: I dont mind if these are Camera and Vision Service, without the Name suffix
```

Reply:
I'm cool with this.
I will have `MotionConfiguration` reference `ObstacleDetectorName` and I will introduce `ObstacleDetector` which the nav service will use.
Since the two will have different use cases, I think `ObstacleDetectorName` should live in `motion.go` while the new `ObstacleDetector` should live in `navigation.go`

Reply:
I like that breakdown